frontend/lib/editor/initiative.ts

```diff
@@ -0,0 +1,174 @@
+  requiresInvitationToJoin: EditorField<Initiative, 'requiresInvitationToJoin'>
+  isAcceptingNewMembers: EditorField<Initiative, 'isAcceptingNewMembers'>
+  isAcceptingNewPortfolios: EditorField<Initiative, 'isAcceptingNewPortfolios'>
+  language: EditorField<Initiative, 'language'>
+  pactaVersion: EditorField<Initiative, 'pactaVersion'>
+  createdAt: EditorField<Initiative, 'createdAt'>
+}
+
+const createEditorInitiative = (initiative: Initiative): EditorInitiative => {
+  return {
+    id: {
+      name: 'id',
+      label: 'ID',
+      isRequired: true,
+      validation: Validation.AlphanumericAndDashesAndUnderscores,
+      originalValue: initiative.id,
+      currentValue: initiative.id,
+    },
+    name: {
+      name: 'name',
+      label: 'Name',
+      isRequired: true,
+      validation: Validation.NotEmpty,
+      originalValue: initiative.name,
+      currentValue: initiative.name,
+    },
+    affiliation: {
+      name: 'affiliation',
+      label: 'Affiliation',
+      originalValue: initiative.affiliation,
+      currentValue: initiative.affiliation,
+    },
+    publicDescription: {
+      name: 'publicDescription',
+      label: 'Public Description',
+      isRequired: true,
+      validation: Validation.NotEmpty,
+      originalValue: initiative.publicDescription,
+      currentValue: initiative.publicDescription,
+    },
+    internalDescription: {
+      name: 'internalDescription',
+      label: 'Internal Description',
+      originalValue: initiative.internalDescription,
+      currentValue: initiative.internalDescription,
+    },
+    requiresInvitationToJoin: {
+      name: 'requiresInvitationToJoin',
+      label: 'Requires Invitation to Join',
+      originalValue: initiative.requiresInvitationToJoin,
+      currentValue: initiative.requiresInvitationToJoin,
+    },
+    isAcceptingNewMembers: {
+      name: 'isAcceptingNewMembers',
+      label: 'Accepting New Members',
+      originalValue: initiative.isAcceptingNewMembers,
+      currentValue: initiative.isAcceptingNewMembers,
+    },
+    isAcceptingNewPortfolios: {
+      name: 'isAcceptingNewPortfolios',
+      label: 'Accepting New Portfolios',
+      originalValue: initiative.isAcceptingNewPortfolios,
+      currentValue: initiative.isAcceptingNewPortfolios,
+    },
+    language: {
+      name: 'language',
+      label: 'Language',
+      isRequired: true,
+      validation: Validation.NotEmpty,
+      originalValue: initiative.language,
+      currentValue: initiative.language,
+    },
+    pactaVersion: {
+      name: 'pactaVersion',
+      label: 'PACTA Version',
+      isRequired: true,
+      validation: Validation.NotEmpty,
+      originalValue: initiative.pactaVersion,
+      currentValue: initiative.pactaVersion,
+    },
+    createdAt: {
+      name: 'createdAt',
+      label: 'Created At',
+      originalValue: initiative.createdAt,
+      currentValue: initiative.createdAt,
+    },
+  }
+}
```

Comment:
I like this part, in the sense that it describes each field and required/validation status, but everything else below is a ton of very error-prone boiler plate. Some ideas:

- Take a look at VeeValidate and see if it's worth using that (or at least borrowing some ideas)
- If not, it _should_ be possible to replace `computedX`, `getComputedXChanges`, and `getComputedIncompleteFields` with just one of each, instead of one-per-entity, by iterating over the object properties and doing more or less what you're doing now.

Reply:
Done the second one - PTAL, and lmk if you're comfortable.